compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1707,6 +1716,33 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
         }
     }
 
+    fn encode_debugger_visualizers(&mut self) -> Lazy<[DebuggerVisualizerFile]> {
+        empty_proc_macro!(self);
+
+        let debugger_visualizers = self
+            .tcx
+            .sess
+            .debugger_visualizers()
+            .iter()
+            .filter_map(|dbgvis| match dbgvis {
+                DebuggerVisualizer::Natvis(path) => {
+                    let bytes = std::fs::read(path.as_path()).unwrap();
+                    let text = std::str::from_utf8(&bytes).unwrap_or("").to_string();
```

Comment:
I think you should `read_to_string` and error out on non-utf-8 contents rather than silently discarding. This is what `read_to_string` already does.